bot/rules/mentions.py

```diff
@@ -1,24 +1,65 @@
 from typing import Dict, Iterable, List, Optional, Tuple
 
-from discord import Member, Message
+from discord import DeletedReferencedMessage, Member, Message, MessageType, NotFound
+
+import bot
+
+log = bot.log.get_logger(__name__)
 
 
 async def apply(
     last_message: Message, recent_messages: List[Message], config: Dict[str, int]
 ) -> Optional[Tuple[str, Iterable[Member], Iterable[Message]]]:
-    """Detects total mentions exceeding the limit sent by a single user."""
+    """
+    Detects total mentions exceeding the limit sent by a single user.
+
+    Excludes mentions that are bots, themselves, or replied users.
+
+    In very rare cases, may not be able to determine a
+    mention was to a reply, in which case it is not ignored.
+    """
     relevant_messages = tuple(
         msg
         for msg in recent_messages
         if msg.author == last_message.author
     )
-
+    # We use msg.mentions here as that is supplied by the api itself, to determine who was mentioned
+    # Additionally, msg.mentions includes the user replied to, even if the mention doesn't occur in the body.
+    # In order to exclude users who are mentioned as a reply, we check if the msg has a reference
+    #
+    # While we could use regex to parse the message content, and get a list of the mentions,
+    # that is very prone to breaking, as both the markdown, if discord used it, code block, etc
     total_recent_mentions = sum(
         not user.bot
+        and msg.author != user
```

Comment:
I think this is more readable than if its moved.

Reply:
I was thinking moreso because the thing you're summing is `not user.bot and msg.author != user`, and so should go all on one line. The "extra" lines in `sum` are for what you're iterating over; not for expanding the condition.

Also, if you do move to one line, then `not (user.bot or msg.author == user)` is better in terms of efficiency, but it's arguably not worth the readability downgrade so will leave up to you.